agent/background.go

```diff
@@ -17,32 +17,26 @@ func (a *Agent) startHeartbeat(ctx context.Context, tc *taskContext, heartbeat c
 	if a.opts.HeartbeatInterval != 0 {
 		heartbeatInterval = a.opts.HeartbeatInterval
 	}
+
 	ticker := time.NewTicker(heartbeatInterval)
 	defer ticker.Stop()
-	failed := 0
+
+	shouldReturn, failed, signalBeat := a.doHeartbeat(ctx, tc, 0)
+	if signalBeat != "" {
+		heartbeat <- signalBeat
+	}
+	if shouldReturn {
```

Comment:
This is redundant. Just return in the `signalBeat != ""` case. There's no need for the boolean.